app/js/components/queryBuilder/queryParamSelector/queryParamSelector.component.js

```diff
@@ -124,14 +133,15 @@ const QueryParamSelectorComponent = {
       /**
        * Filter our results so we don't return what's already added to the query
        * @param  {Array} rawSearchResults - array of objects returned from search
+       * @param {Array} addedSet - array of selected objects added to query
        *
-       * @return {Array} array of results filtered by the current state state from the query
+       * @return {Array} array of results filtered by the current state from the query
        */
-      let _filteredSearchResults = rawSearchResults => {
+      let _filteredSearchResults = (rawSearchResults, addedSet) => {
         let comparator = vm.currentState() == "mutations" ? "_id" : "acronym";
         return $filter("notInArrayFilter")(
           rawSearchResults,
-          vm[`${vm.currentState()}Set`],
+          addedSet,
```

Comment:
💯 This is an excellent refactor. Explicit parameters are usually a better approach.